### PR TITLE
feat: save user details

### LINKED ISSUE #4: Save User details to Database
- Remove Console log in Twitter.js 
- Import UserSchema and UserAccessToken Schema in Twitter.js
- Check if user exists in the database using twitterId (which should be in the 'profile' object returned by twitter)
- If user exists in the database call cb(null,user) => user is the existing user from the db converted to json
- if user does not exists in the database , create and save the new user and call cb(null,user) => user is the newly created user in json
All this should be done in Twitter.js in the place of the console.log

### PR DESCRIPTION
**Pull request by Dun-sin**
## changes proposed
- save users to database

### CODE REVIEW

File: src/PassportStrategies/Twitter.js [hunk 1]
```diff
@@ -12,7 +14,12 @@ export default function (passport) {
                     'https://api.twitter.com/1.1/account/verify_credentials.json?include_email=true',
             },
             async (token, tokenSecret, profile, cb) => {
-                cb(null, profile);
+                UserModel.findById({twitterId: profile.id}, (error, user) => {
```
Comment (mathiasayivor) [inline src/PassportStrategies/Twitter.js:17]: Why not `UserModel.findOne`?
Also, you need to pass the rest of the data when registering; currently, you're only passing twitterId.

**Review by mathiasayivor — CHANGES_REQUESTED** [review]

**Comment by A-BMT02** [comment]
Hey @Dun-sin Just a couple of corrections

- If findById returns error , it doesnt mean that the user does not exists. It is usually a connection problem or something like that
- I suggest you wrap everyting in a try-catch block so it will be easier to handle errors later on
- To check if the user exists you could do something like this
```
try {
const userExists = await UserModel.findOne({twitterId: profile.id})
if(userExists) {
    return cb(null, userExists.lean())
} else {
  //user doesnt exists
  //Here you should create the 'UserModel' document based on the schema in /src/Models/UserModel
  //It would look something like this
  cons newUser = UserModel.create({
  twitterInfo : {
      id : profile.id
      username : profile.something , 
     displayName : profile.something , 
     pictureSrc : profil.something
  }, 
  registeredAt : (Generate current date)
  })
  //after saving the user , you should also save the tokens in the UserAccessTokenModel 
  //same thing as above but use the schema in src/Models/UserAccessTokenModel
  ...
return cb(null , newUser.Json())
}
} catch (error) {
//we'll make error handling a separate issue , for now just call cb(error)
return cb(error) //passport handles all our errors , if there is any error in the try block , the 'failureRedirect' (line 33 in index.js) will be called
}
```

File: src/PassportStrategies/Twitter.js (hunk 1 above)
Comment (Dun-sin) [reply src/PassportStrategies/Twitter.js:17]: Id is unique right? That's why I didn't bother with the rest I guess

Comment (A-BMT02) [reply src/PassportStrategies/Twitter.js:17]: I think he meant when creating the new user , you should save the rest of the data too 
He didnt mean you should use all the data to find the user  ,for that the twitter id is enough

Comment (Dun-sin) [reply src/PassportStrategies/Twitter.js]: oh yeah that makes sense

File: src/PassportStrategies/Twitter.js [hunk 2]
```diff
@@ -12,15 +24,29 @@ export default function (passport) {
...
-                    profile
-                );
-                cb(null, profile);
+               try{
+                const userExists = await UserModel.findOne({twitterId: profile.id})
+
+                if(userExists) {
+                    return cb(null, userExists.lean())
+                } else {
+                    const newUser = UserModel.create({
```
Comment (A-BMT02) [inline src/PassportStrategies/Twitter.js:33]: When creating the newUser , save all the rest of the data too , not just the twitterId

Comment (Dun-sin) [reply src/PassportStrategies/Twitter.js:33]: yeha got that

**Pull request closed without merging**